sql/core/src/main/scala/org/apache/spark/sql/execution/sort.scala

```diff
@@ -78,6 +79,11 @@ case class ExternalSort(
       val sorter = new ExternalSorter[InternalRow, Null, InternalRow](ordering = Some(ordering))
       sorter.insertAll(iterator.map(r => (r.copy(), null)))
       val baseIterator = sorter.iterator.map(_._1)
+      val context = TaskContext.get()
+      context.taskMetrics().incDiskBytesSpilled(sorter.diskBytesSpilled)
```

Comment:
It's confusing. The sorter only does it if you call `writePartitionedFiles`, not when you drain the iterator.